### PR TITLE
Move PawnsOnSquares out of Pawns

### PR DESCRIPTION
**Pull request by protonspring**
This is a non-functional simplification which removes PawnsOnSquares from pawns.
Adds a pieces_on_same_color() instead of pawns_on_same_color() so we don't lose any abstraction.

STC
LLR: 2.96 (-2.94,2.94) [-3.00,1.00]
Total: 15926 W: 3544 L: 3411 D: 8971
http://tests.stockfishchess.org/tests/view/5c3a68cd0ebc596a450cb732

### CODE REVIEW

**Comment by mcostalba** [comment]
I find the helper function particularly confusing. I'd prefer to explicitly state what we are doing in the code:

```c++
auto sqares = (DarkSquares & s) ? DarkSquares : ~DarkSquares;
score -= BishopPawns * popcount(pos.pieces(Us, PAWN) & squares);

```

**Comment by protonspring** [comment]
I would like that better as well and I did that in an earlier test.
@snicolet didn't like losing the function which is a convenient function
for future developers.

**Comment by mcostalba** [comment]
hmmm usually first you have a problem (duplicated code) then you fix it (write a single helper). Inverting the order is not a good idea in software, although very common because it is usually confused with writing generic code when possible, but the 2 things are indeed different.

Anyhow it's up to @snicolet to decide what he prefers.

**Comment by protonspring** [comment]
I agree with that in principle.  However, I have not removed a helper function here, just changed it so that it could be used more generally.  The main benefit here is the removal of memory arrays and code that do not provide any real benefit.  This reduces the size of a PawnEntry as well as the memory usage of sf.

**Comment by protonspring** [comment]
Darn. . I just tried to update master, but this doesn't look right. sry.

File: src/bitboard.h [hunk 1]
```diff
@@ -395,4 +395,8 @@ inline Square pop_lsb(Bitboard* b) {
 inline Square frontmost_sq(Color c, Bitboard b) { return c == WHITE ? msb(b) : lsb(b); }
 inline Square  backmost_sq(Color c, Bitboard b) { return c == WHITE ? lsb(b) : msb(b); }
 
+inline int pieces_on_same_color(Bitboard pieces, Square s) {
```
Comment (ddugovic) [inline src/bitboard.h]: Perhaps if re-writing this, create `inline Bitboard color_complex(Square s)` instead?

**Comment by protonspring** [comment]
Sry, i do not understand your comment.

**Comment by ddugovic** [comment]
I'm just saying, a method which returns a `Bitboard` instead of an `int` could be used for other purposes (such as bitwise-and with my pawns `Bitboard` to determine if it's a good bishop or a bad bishop, or bitwise-and with the opponent's bishop[s] to check for same/opposite-colored bishops).

**Pull request closed without merging**